frontend/src/components/SearchProducts.js

```diff
@@ -0,0 +1,61 @@
+import styled from 'styled-components/macro'
+import {useState} from 'react'
+import {getProductsById, postProduct} from "../service/ApiService";
+
+
+export default function SearchProducts({ products, setProducts }) {
+    const [search ,setSearch] = useState("")
+
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        getProductsById(search)
+            .then(response =>{
+                postProduct(response).then( updatedProduct => setProducts(
+                    [...products, updatedProduct]
```

Comment:
it looks like you are appending search products to your existing list - I would have expected the search result to replace the current list
you may also want to replace the search term after the search has been triggered
`setSearch('')`